searx/engines/dailymotion.py

```diff
@@ -24,21 +25,32 @@
 
 # search-url
 # see http://www.dailymotion.com/doc/api/obj-video.html
-search_url = 'https://api.dailymotion.com/videos?fields=created_time,title,description,duration,url,thumbnail_360_url,id&sort=relevance&limit=5&page={pageno}&{query}'  # noqa
-supported_languages_url = 'https://api.dailymotion.com/languages'
+search_url = 'https://api.dailymotion.com/videos?fields=created_time,title,description,duration,url,thumbnail_360_url,id&sort=relevance&limit=5&page={pageno}&{query}'
+
+# The request query filters by 'localization', therefore instead of fetching
+# languages we need to fetch locales.
+supported_languages_url = 'https://api.dailymotion.com/locales'
 
 
-# do search-request
 def request(query, params):
-    if params['language'] == 'all':
-        locale = 'en-US'
-    else:
-        locale = match_language(params['language'], supported_languages)
 
+    language = params['language']
+    if language == 'all':
+        language = 'en-US'
+
+    locale = babel.Locale.parse(language, sep='-')
```

Comment:
With this version, `supported_languages` is ignored.
Do you intended to remove `match_language`?

For example `ca` (Catalan)
* becomes `en` with the `match_language`
* remains `ca` with this line --> I don't know how dailymotion deals with this unexpected value.

Reply:
> Do you intended to remove `match_language`?

May be .. while I review engine by engine I see that this function does not fit to the language that is needed for the query send to the search engine.

> * remains `ca` with this line --> I don't know how dailymotion deals with this unexpected value.

Oops, my fail .. I will fix it. But this will not improve the result list:

> In contrast to the documentation of the api [1], dailymotion's API does not care
about a 'localization' argument or a 'language' filter[2] (may be there is an
API account needed to get in use of this options).

Reply:
I ask the here, but I see the same logic in #1073
Do you intend to send the language & region unfiltered to the engines?
I think it is regression (`match_language` is going to map `ca-ES` to `ca` or `en`, so there is no unexpected value).

Reply:
We can't send `ca` to a engine that only except `ca_ES`.

Reply:
So we need a mapping between the language preference and the engine expected value.
* In the master branch, the `match_language` function maps the user preference to the engine expected values.
* In the PR, the mapping is done inside the engine code. However it may send unexpected value: `ca`or `ca_ES` are neither accepted by Dailymotion (see comment below).

> I see that this function does not fit to the language that is needed for the query send to the search engine.

Can you say more and/or give examples?

---

Note: I don't have a clear idea what is the solution.

Reply:
> Can you say more and/or give examples?

E.g. bing, the script Hant & Hans

```py
>>> import babel
>>> from searx import settings
>>> from searx.engines import load_engines
>>> engines = load_engines(settings['engines'])
>>> bing = engines['bing']
>>> bing.supported_languages
['af', 'am', 'ar', 'as', 'az-latn', 'be', 'bg', 'bn', 'bs-latn', 'ca', 'ca-es-valencia', 'chr-cher', 'cs', 'cy', 'da', 'de', 'el', 'en', 'es', 'et', 'eu', 'fa', 'fi', 'fil', 'fr', 'ga', 'gd', 'gl', 'gu', 'ha-latn', 'he', 'hi', 'hr', 'hu', 'hy', 'id', 'ig', 'is', 'it', 'ja', 'ka', 'kk', 'km', 'kn', 'ko', 'kok', 'ku-arab', 'ky', 'lb', 'lo', 'lt', 'lv', 'mi', 'mk', 'ml', 'mn-Cyrl-MN', 'mr', 'ms', 'mt', 'nb', 'ne', 'nl', 'nn', 'nso', 'or', 'pa-arab', 'pa-guru', 'pl', 'prs', 'pt-BR', 'pt-PT', 'quc', 'quz', 'ro', 'ru', 'rw', 'sd-arab', 'si', 'sk', 'sl', 'sq', 'sr-cyrl', 'sr-latn', 'sv', 'sw', 'ta', 'te', 'tg-cyrl', 'th', 'ti', 'tk', 'tn', 'tr', 'tt', 'ug', 'uk', 'ur', 'uz-latn', 'vi', 'wo', 'xh', 'yo', 'zh-Hans', 'zh-Hant', 'zu']
>>> 
>>> bing.match_language('zh', bing.supported_languages, bing.language_aliases)
'zh-Hans'
>>> babel.Locale.parse('zh', sep = '-')
Locale('zh')
>>> 
>>> bing.match_language('zh-CN', bing.supported_languages, bing.language_aliases)
'zh-Hans'
>>> babel.Locale.parse('zh-CN', sep = '-')
Locale('zh', territory='CN', script='Hans')
>>> 
>>> bing.match_language('zh-TW', bing.supported_languages, bing.language_aliases)
'zh-Hans'
>>> babel.Locale.parse('zh-TW', sep = '-')
Locale('zh', territory='TW', script='Hant')
>>> 
>>> bing.match_language('zh-HK', bing.supported_languages, bing.language_aliases)
'zh-Hans'
>>> babel.Locale.parse('zh-HK', sep = '-')
Locale('zh', territory='HK', script='Hant')

```